### PR TITLE
Decouple chipsec_main from logger

### PR DESCRIPTION
**Pull request by BrentHoltsclaw**
Remove start_test from logger
Enhance TestCase class to add start_test funtionality Change logic within run_loaded modules to use TestCase opposed to logger Add fstring to testcase.py
Add Type information to testcase.py

Signed-off-by: brentholtsclaw <brent.holtsclaw@intel.com>

### CODE REVIEW

File: chipsec/testcase.py [hunk 1]
```diff
@@ -174,44 +199,45 @@ def xml_summary(self):
...
-        xml_element = ET.Element("testsuites")
+    def xml_full(self, name) -> str:
+        xml_element = ET.Element('testsuites')
         summary = self.order_summary()
-        summary_dict = dict()
+        summary_dict = {}
         for k in summary.keys():
             if k == 'total':
-                summary_dict[k] = "{:d}".format(summary[k])
+                summary_dict[k] = '{summary[k]:d}'
```
Comment (frinzell) [inline chipsec/testcase.py:218]: Needs to be an fstring

File: chipsec/testcase.py [hunk 2]
```diff
@@ -174,44 +199,45 @@ def xml_summary(self):
...
 
-    def json_full(self):
+    def json_full(self) -> str:
         summary = self.get_results()
         js = json.dumps(summary, sort_keys=False, indent=2, separators=(',', ': '))
         return js
 
-    def xml_full(self, name):
-        xml_element = ET.Element("testsuites")
+    def xml_full(self, name) -> str:
```
Comment (frinzell) [inline chipsec/testcase.py:212]: `name` needs a type hint

File: chipsec/testcase.py [hunk 3]
```diff
@@ -55,34 +57,57 @@ class ExitCode:
...
+        self.startTime = None
+        self.endTime = None
+        self.time = None
+
+    def get_fields(self) -> Dict[str, str]:
+        return {'name': self.name, 'output': self.output, 'result': self.result}
+
+    def start_module(self) -> None:
+        """Displays a banner for the module name provided."""
+        text = f'\n[*] running module: {self.name}'
```
Comment (frinzell) [inline chipsec/testcase.py:76]: Please preserve 'Running' chaptalization.  `[*] Running module:`

**Review by frinzell — CHANGES_REQUESTED** [review]

File: chipsec/testcase.py [hunk 4]
```diff
@@ -55,34 +57,57 @@ class ExitCode:
...
+        self.desc = self.name
+
+    def end_module(self, result: str, arg: str) -> None:
+        self.result = result
+        self.argv = arg
+        self.endTime = time.time()
+        self.time = self.endTime - self.startTime
+
+
+class ChipsecResults():
```
Comment (npmitche) [inline chipsec/testcase.py:88]: empty parentheses.

**Review by npmitche — APPROVED** [review]

**Comment by jun2zhou** [comment]
@BrentHoltsclaw  I compare the log with previous version, the alignment and color changed.
image

**Comment by npmitche** [comment]
> @BrentHoltsclaw I compared the log with previous version, the alignment and color changed. image

I'm not seeing this:
Screenshot 2023-01-18 090710
Screenshot from 2023-01-18 08-41-30

 Though I am seeing an issue with logger().error() that I'll need to look into:
```
WARNING: NOT APPLICABLE: chipsec.modules.common.smm
[CHIPSEC] Modules with exceptions 2:
Traceback (most recent call last):
  File "C:\Users\Nathaniel\Desktop\chipsec-loggingchanges\chipsec_main.py", line 489, in <module>
    sys.exit(main())
  File "C:\Users\Nathaniel\Desktop\chipsec-loggingchanges\chipsec_main.py", line 484, in main
    return chipsecMain.main()
  File "C:\Users\Nathaniel\Desktop\chipsec-loggingchanges\chipsec_main.py", line 472, in main
    modules_failed = self.run_all_modules()
  File "C:\Users\Nathaniel\Desktop\chipsec-loggingchanges\chipsec_main.py", line 373, in run_all_modules
    return self.run_loaded_modules()
  File "C:\Users\Nathaniel\Desktop\chipsec-loggingchanges\chipsec_main.py", line 343, in run_loaded_modules
    results.print_summary(runtime)
  File "C:\Users\Nathaniel\Desktop\chipsec-loggingchanges\chipsec\testcase.py", line 293, in print_summary
    logger().error(mod)
AttributeError: 'Logger' object has no attribute 'error'
```

**Comment by BrentHoltsclaw** [comment]
@jun2zhou can you provide more information on how you created this?  I have not been able to create a similar error.

File: chipsec/testcase.py [hunk 5]
```diff
@@ -232,53 +259,51 @@ def markdown_full(self, name):
...
+                logger().log(f'[CHIPSEC] Modules {k:16}{summary[k]:d}')
+            elif k == 'warnings':
+                logger().log(f'[CHIPSEC] Modules with {k:11}{len(summary[k]):d}:')
+                for mod in summary[k]:
+                    logger().log_warning(mod)
+            elif k == 'exceptions':
+                if len(summary[k]) > 0:
+                    logger().log(f'[CHIPSEC] Modules with {k:11}{len(summary[k]):d}:')
+                    for mod in summary[k]:
+                        logger().error(mod)
```
Comment (npmitche) [inline chipsec/testcase.py:293]: should be `logger().log_error(mod)

**Review by npmitche — CHANGES_REQUESTED** [review]

**Review by npmitche — APPROVED** [review]

**Review by dscott90 — APPROVED** [review]

**Pull request merged**